### PR TITLE
Task/73 refreshing auhentication (#98)

### PR DESCRIPTION
**Pull request by storkandstars**
# Overview

Implementing a http client that can intercept http requests and responses and use our authenticated strategy. Fot that reason we use a http client generator called [Chopper](https://hadrien-lejard.gitbook.io/chopper). We have two http clients. One is public and adds api key to all request. Another is secured that adds a bearer token to the request and implements refreshing strategy via custom Authenticator.

There are implemented following services which handle its endpoint and call methods:

- Sign Up
- Sign In
- Password reset
- Password change
- Refreshing access token (its part of Authenticator)

# Test cases:
- Try to sign in with an existing account. It should login and stay logged after reopening the application
- Try to sign in with a non-existing account. It should not passed. There is no error handeling for UI at this moment.
- Try to reset a password. It should send a email with password reset link. 
- Try to Sign up. Verify your email. And sign in. You should enter the dashboard.

### CODE REVIEW

File: lib/data/authenticators/refresh_token_authenticator.dart [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+        return request.copyWith(headers: {
+          ...request.headers,
+          "Authorization": "${authToken.tokenType} ${authToken.accessToken}",
+        });
+      }
+
+      return null;
+    } catch (e) {
+      return null;
```
Comment (jvrastil) [inline lib/data/authenticators/refresh_token_authenticator.dart:36]: is this correct? Should't it pass exception from `_refreshToken` method?

File: lib/data/http_clients/public_http_client.dart [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import 'package:simplio_app/data/http_clients/http_client.dart';
+import 'package:simplio_app/data/interceptors/api_key_interceptor.dart';
+import 'package:simplio_app/data/http_clients/json_serializable_convertor.dart';
+import 'package:simplio_app/data/services/password_reset_service.dart';
+import 'package:simplio_app/data/services/sign_in_service.dart';
+import 'package:simplio_app/data/services/sign_up_service.dart';
+import 'package:simplio_app/data/services/refresh_token_service.dart';
+
+/// Public Http Client provides with http endpoints which can be called
+/// and do not need Authorization with Bearer token.
```
Comment (jvrastil) [inline lib/data/http_clients/public_http_client.dart:11]: *does not need

**Review by jvrastil — APPROVED** [review]

File: lib/data/authenticators/refresh_token_authenticator.dart (hunk 1 above)
Comment (storkandstars) [reply lib/data/authenticators/refresh_token_authenticator.dart]: Removing try catch block

File: lib/data/http_clients/public_http_client.dart (hunk 2 above)
Comment (storkandstars) [reply lib/data/http_clients/public_http_client.dart]: Removed obsolete comments

**Pull request merged**